### ISSUE TITLE
Tree-style navgiation

### ISSUE THREAD

**Issue by jeenuv** [ranger/ranger#1612]
More of a feature request: I wondered if there are any plans for offering tree-style navigation of the file system rather than Miller-style columns.

I use Ranger as my file picker from Vim. Most IDEs offer a tree-style file navigator as a side bar, which is good to simultaneously view contents of multiple directories and their hierarchy.

**Comment by toonn** [comment]
Sounds like a custom rendering of `:flat`, sort of.

**Comment by jeenuv** [comment]
`:flat` works, as you suggested.

It'd be more pleasing if we could drop the common prefix on the files fetched from directories underneath, and:
* Indent them proportionately to their depth, or
* replace them with `'-` to make them stand out.

**Comment by toonn** [comment]
Yeah, that's why I said "custom rendering," I actually think this may be a more pleasing version of `:flat`, we could use a `:tree` alias of `:flat` that uses some unicode symbols:
```
dir
├─ fileA (this would replace "dir/fileA")
└─ fileB
```

I do think this could have some repercussions on sorting a flattened directory though, so it'll either be slightly janky or take a lot of work.

**Comment by toonn** [comment]
Currently the sorting happens based on the entire path name. If we change the name using some symbols for display like those unicode things then all the `├─` files might be sorted together and then all the `└─` files. That's obviously not what you'd want. Or at least it doesn't sound very useful to me.

**Comment by zyklotomic** [comment]
If this were to be implemented, should this be in a completely separate "tree" viewmode, or was there something else that you guys were envisioning.

**Comment by toonn** [comment]
What I described would be a very light hack on the existing `:flat` command.

**Comment by elig0n** [comment]
If I may suggest then something akin to https://www.han.de/~werner/ytree.html 
where directories are shown on the tree in the upper portion while files show up in the bottom
part.

**Comment by BigBrotherJu** [comment]
3 years later, any updates on this?

**Comment by zyklotomic** [comment]
Sorry! Unfortunately, i never got around to this 😞

**Comment by bbottema** [comment]
What about now?
